internal/consumer/consumer.go

```diff
@@ -388,6 +380,31 @@ func (i *InventoryConsumer) ProcessMessage(headers map[string]string, relationsE
 	return "", nil
 }
 
+// calculateTuplesOrNoOp parses the message, calculates tuples, and checks if it's a no-op.
+// Returns the tuples, a boolean indicating if it's a no-op (empty), and an error.
+func (i *InventoryConsumer) calculateTuplesOrNoOp(msgValue []byte, operation string, operationType biz.EventOperationType, txid string) (model.TuplesToReplicate, bool, error) {
+	tupleEvent, err := ParseMessage(msgValue, operation)
+	if err != nil {
+		metricscollector.Incr(i.MetricsCollector.MsgProcessFailures, "ParseMessage")
+		i.Logger.Errorf("failed to parse message for tuple: %v", err)
+		return model.TuplesToReplicate{}, false, err
+	}
+
+	tuplesToReplicate, err := i.SchemaService.CalculateTuples(*tupleEvent, operationType)
+	if err != nil {
+		metricscollector.Incr(i.MetricsCollector.MsgProcessFailures, "CalculateTuples")
+		i.Logger.Errorf("failed to calculate tuples: %v", err)
+		return model.TuplesToReplicate{}, false, err
+	}
+
+	isNoOp := tuplesToReplicate.IsEmpty()
+	if isNoOp {
+		i.Logger.Infof("no-op for operation=%s, txid=%s", operation, txid)
+	}
+
+	return tuplesToReplicate, isNoOp, nil
+}
+
```

Comment:
Nit: Probably a bit redundant/risky to return `isNoOp` if it always aligns with `tuplesToReplicate` being empty (or populated). Callers could just use `tuplesToReplicate.IsEmpty()` as the source of truth.